### DISCUSSION TITLE
How to use Auth API - I'm getting Errors.Token.Invalid

### DISCUSSION THREAD

**Discussion by Yberion** [zitadel/zitadel#7396] [Q&A] [answered]
Hello,

I'm trying to use the [Auth API](https://zitadel.com/docs/apis/resources/auth).

I'm currently login with such url `http://192.168.0.30:8080/ui/login/login?authRequestID=254067341732610050`.
It is an `User Agent` with Code Flow authentication.

I'm getting the `access_token` with `http://192.168.0.30:8080/oauth/v2/token` (I'm using [this lib](https://github.com/damienbod/angular-auth-oidc-client) for the authentication, so I have everything in my `Session Storage`).

When I use the `access_token` as `Bearer` returned by `/oauth/v2/token` let's say for a `GET` to `/auth/v1/users/me`, I'm getting an `Errors.Token.Invalid`.

When I'm using the token from the session storage `zitadel:access_token` from the UI Console (http://192.168.0.30:8080/ui/console/users/me) then it works fine.

How am I supposed to use those APIs with oauth Token?

I'd like to make my own Profile page.

I'm new to Zitadel, sorry if this is in the documentation, I've tried to read it but probably missed/didn't understand the info.

**Comment by hifabienne** [comment, marked as answer]
Hei @Yberion 
If you want to access the ZITADEL APIs you have to ensure that the audience for ZITADEL is included.
You can do this by including the following scope: urn:zitadel:iam:org:project:id:zitadel:aud

If you want to read more about it you can find it here: https://zitadel.com/docs/guides/integrate/access-zitadel-apis#request-an-oauth-token-with-audience-for-zitadel

**Reply by Yberion** [reply]
Hello @hifabienne 

Thanks for the fast answer, adding the scope fixed the problem! 😄
